tests/comdb2sys.test/testtransactionstate.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+a_dbn=$1
+default=local
+
+setup () {
+cdb2sql ${CDB2_OPTIONS} $a_dbn local - <<EOF
+create table txtest(a int); \$\$
+put tunable debug.txn_sleep = 5;
+insert into txtest(a) values(1);
+EOF
+}
+
+setup &
+sleep 1
+cdb2sql ${CDB2_OPTIONS} $a_dbn local 'select * from comdb2_transaction_state' >> testtransactionstate.out
+sleep 10
+cdb2sql ${CDB2_OPTIONS} $a_dbn local 'select * from comdb2_transaction_state' >> testtransactionstate.out
+
+nbusy=$(grep -o 'busy' testtransactionstate.out | wc -l)
+nidle=$(grep -o 'idle' testtransactionstate.out | wc -l)
+rc=$(($nbusy==0 || $nidle==0 ? 1 : 0)) # There should be at least one busy thread and one idle thread.
+
+if [[ $rc -ne 0  ]]; then
+    echo "Failed transaction state test"
+    exit $?
```

Comment:
How is the script invoked? Also `$?` here would be the return code of the `echo` command above, which would most likely be zero. Did you mean to use `$rc` here?

Reply:
[Updated `runit`](https://github.com/bloomberg/comdb2/pull/3547/files#diff-653d54fb85eed9304c6415b6705e0bc7f69a15770581e5cdd36ce3a44ca9204cR150), [corrected return code](https://github.com/bloomberg/comdb2/pull/3547/files#diff-144f6c6e4a1af734e0fe42ca29d82dd4c14bd7be049fd9069423207368f7e452R27), and modified the test script so that requests are routed to master where appropriate.

Reply:
Also moved debug sleep to [bdb_tran_begin_ll_int](https://github.com/bloomberg/comdb2/pull/3547/files#diff-836ea607bf1bb0babe1e645404e85850b7949f391f16972d84db77eb53cf3dc7R1041) where it should be invoked consistently.